### PR TITLE
Adds method descriptions from odkCommon.js to docs

### PR DESCRIPTION
**Pull request by bryce-p-williams**

### CODE REVIEW

**Comment by linl33** [comment]
Some comments for the changes,

1. Currently the only way to access this page is through search. This page will need to be linked from another page or added to the sidebar.
2. It is unclear from the title or description that these methods are in `odkCommon`.
3. The table formatting is not very easy to read. Maybe use paragraphs instead?
4. It would be easier to read and find information if description of the return value and parameters are separated from the description for the function.

Some functions in `odkCommon` are missing:
 - isString
 - extractLangOnlyLocale
 - i18nFieldNames (not a function)
 - has*Localization (total of 5)

Some things that would be nice to have, these can be addressed by a future pull request:
 - A note on each function whether the call goes over the Android WebView bridge
 - Better formatting for code snippets, class, function, variables, etc. and other text that should have special formatting.

File: odkx-src/injected-interfaces-methods.rst [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+      into a url by which it can be accessed.
+  * - | lookupToken(stringToken)
+    - Return the content of a display object for the given token. Note that this might
+      include text, hint, image, etc. that are then localizable. In general, the resulting
+      object can be customized further in survey XLSX files by specifying overrides for these
+      fields.
+  * - | getPreferredLocale()
+    - Return an object representing the locale that was configured by the user in
+      the Java-side's Device Settings
+  * - | getLocaleDetails
```
Comment (linl33) [inline odkx-src/injected-interfaces-methods.rst]:
```suggestion
  * - | getLocaleDetails()
```

File: odkx-src/injected-interfaces-methods.rst [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    - Get an object containing details about the preferred locale (preferredLocale), whether
+      the preferred locale is the same as the Device's locale (usingDeviceLocale), and other
+      information about the device locale (isoCountry, displayCountry, isoLanguage, displayLanguage)
+  * - | hasLocalization(locake, i18nToken)
```
Comment (linl33) [inline odkx-src/injected-interfaces-methods.rst]:
```suggestion
  * - | hasLocalization(locale, i18nToken)
```

File: odkx-src/injected-interfaces-methods.rst [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+  * - | hasListener()
+    - Returns true if there is a listener already registered.
+  * - | getPlatformInfo()
+    - Returns the platform info as a stringified JSON object
+      containing the keys: container, version, appName, baseUri, and logLevel.
+  * - | getFileAsUrl(relativePath)
+    - Take the path of a file relative to the app folder and return an absolute url
+      by which it can be accessed.
+  * - | getRowFileAsUrl(tableId, rowId, rowPathUri)
+    - convert the rowpath value for a media attachment (e.g. uriFragment) field
```
Comment (linl33) [inline odkx-src/injected-interfaces-methods.rst]:
```suggestion
    - Convert the rowpath value for a media attachment (e.g. uriFragment) field
```

File: odkx-src/injected-interfaces-methods.rst [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+      under screen rotations. Useful if browser cookies don't work.
+  * - | getSessionVariable(elementPath)
+    - Retrieve a persistent key-value. This lasts for the duration of this screen and is retained
+      under screen rotations. Useful if browser cookies don't work.
+  * - | genUUID()
+    - Generate a globally unique id.
+  * - | constructSurveyUri(tableId, formId, rowId, screenPath, elementKeyToValueMap)
+    - Constructs a uri of the form "content://org.opendatakit.provider.forms/<appName>/<tableId>
+      /<formId>/#instanceId=<rowId>&screenPath=<screenPath>" followed by "&<key>=<value>" for each
+      key in the elementKeyToValueMap).
```
Comment (linl33) [inline odkx-src/injected-interfaces-methods.rst]:
```suggestion
      key in the elementKeyToValueMap.
```

**Comment by linl33** [comment]
The new version looks good. There are some style issues that will eventually get caught by the build server, you could start addressing those.

See the README for the make target for the checks.

**Comment by wbrunette** [comment]
Closing PR in favor of pull request #1179

**Pull request closed without merging**